lib/topological_inventory/core/service_catalog_client.rb

```diff
@@ -0,0 +1,59 @@
+require "rest_client"
+require "topological_inventory/core/service_plan_client"
+
+module TopologicalInventory
+  module Core
+    class ServiceCatalogClient
+      def initialize(source)
+        @source = source
+      end
+
+      # TODO? Move this to a module in the API repo
+      def order_service_plan(plan_name, service_offering_name, additional_parameters)
+        payload = ServicePlanClient.new.build_payload(
+          plan_name, service_offering_name, catalog_parameters(additional_parameters)
+        )
+        response = make_request(method: :post, url: order_service_plan_url, payload: payload)
+        JSON.parse(response.body)
+      end
+
+      private
+
+      def base_url
+        URI.join(@source.default_endpoint.host, "apis/servicecatalog.k8s.io/v1beta1/").to_s
```

Comment:
What about the rest of the URI components?

Reply:
What do you mean? Or do you mean I should combine `base_url` and `order_service_plan_url` since nothing else uses `base_url`?

Reply:
`Endpoint.host` would be something like `"example.com"`.  This doesn't take into account the rest of the URL components that we store on the `Endpoint` for example:
- The OpenShift / Kubernetes api usually is not running on port 80/443
- HTTP vs HTTPS
- Any deviations to the base path